internals/notifier.py

```diff
@@ -254,6 +254,113 @@ def get_feature_starrers(self, feature_id):
     return user_prefs
 
 
+class FeatureAccuracyHandler(basehandlers.FlaskHandler):
+  SEND_NOTIFICATIONS = False
+
+  CHROME_RELEASE_SCHEDULE_URL = (
+'https://chromiumdash.appspot.com/fetch_milestone_schedule')
+  ACCURACY_AS_OF_WEEKS = 4
+  MILESTONE_FIELDS = (
+    'dt_milestone_android_start',
+    'dt_milestone_desktop_start',
+    'dt_milestone_ios_start',
+    'dt_milestone_webview_start',
+    'ot_milestone_android_start',
+    'ot_milestone_desktop_start',
+    'ot_milestone_webview_start'
+  )
+  TEMPLATE_PATH = 'accuracy_notice_email.html'
+   
+  def get_template_data(self):
+    """Sends notifications to users requesting feature updates for accuracy."""
+
+    # Do not send notifications
+    # TODO(danielrsmith): This should be removed when the new page to
+    # verify data accuracy is implemented.
+    if not self.SEND_NOTIFICATIONS:
+      return {'message': '0 emails sent or logged.'}
+
+    features_to_notify = self._determine_features_to_notify()
+    email_tasks = self._build_email_tasks(features_to_notify)
+    return self._send_notifications(email_tasks)
+    
+  def _determine_features_to_notify(self):
+    # 'current' milestone is the next stable milestone that hasn't landed.
+    # If reminding 8 weeks in advance, we should only need to notify
+    # for current and (current + 1) milestones.
+    try:
+      resp = requests.get(f'{self.CHROME_RELEASE_SCHEDULE_URL}?mstone=current')
+    except requests.RequestException as e:
+      raise e
+    mstone_info = json.loads(resp.text)
+    mstone = int(mstone_info['mstones'][0]['mstone'])
+    
+    features = models.Feature.get_all()
+    features_to_notify = []
+    now = datetime.now()
+    accuracy_as_of_delta = timedelta(weeks=self.ACCURACY_AS_OF_WEEKS)
+    for feature in features:
+      accuracy_as_of_date = None
+      # Format accuracy date to compare dates.
+      if 'accurate_as_of' in feature and feature['accurate_as_of'] is not None:
+        accuracy_as_of_date = datetime.strptime(
+          feature['accurate_as_of'], '%y-%m-%d %H:%M:%S.%f')
+      # If the data has been recently verified as accurate, no need for email.
+      if (accuracy_as_of_date is not None and
+          accuracy_as_of_date + accuracy_as_of_delta < now):
+        continue
+      # Check each milestone field and see if it corresponds with
+      # the next 2 milestones.
+      for mstone_field in self.MILESTONE_FIELDS:
+        if mstone_field in feature and feature[mstone_field] == mstone:
+          feature['notify_mstone'] = str(mstone)  # milestone for email.
+          features_to_notify.append(feature)
+          break
+        if mstone_field in feature and feature[mstone_field] == mstone + 1:
+          feature['notify_mstone'] = str(mstone + 1)  # milestone for email.
+          features_to_notify.append(feature)
```

Comment:
I think this would allow a case where a feature is added to the list twice (or if you used two lists, added to both).  We should probably check if the feature is already in the list.

Also, the line above would cause such a feature to be notified for mstone +1, however it is probably better for the email to mention mstone since that is coming up sooner.  This problem is avoided if you use two lists and avoid appending if the feature is already in the first list.

Reply:
I've made a change to check all the possible starts to find the nearest milestone that would trigger a notification. Maybe you can take a look and see if it's an effective change?